#** R-Script-for-DEGs-identification-
Its a R Script to find Differentially Expressed Gene from Transcriptomics data 
library(GEOquery)
untar("GSE25861_RAW.tar", exdir = "data")
cels = list.files("data/", pattern = "[gz]")
sapply(paste("data", cels, sep = "/"), gunzip)
phenodata = matrix(rep(list.files("data"), 2), ncol =2)
class(phenodata)
phenodata <- as.data.frame(phenodata)
colnames(phenodata) <- c("Name", "FileName")
phenodata$Targets <- c ("AGA", "IUGR", "IUGR", "IUGR", "IUGR", "IUGR", "AGA", "IUGR", "AGA", "AGA")
write.table(phenodata, "data/phenodata.txt", quote = F, sep = "\t", row.names = F)
-------------------------------------------------------------------------**
library(simpleaffy)
celfiles <- read.affy(covdesc = "phenodata.txt", path = "data")
boxplot(celfiles)
png("box_plot.png")
boxplot(celfiles)
library(RColorBrewer)
cols = brewer.pal(3, "Set1")
eset1 <- exprs(celfiles)
samples <- celfiles$Targets
boxplot(celfiles, col = cols, las = 2)
require(simpleaffy)
require(affyPLM)
celfiles.rma = rma(celfiles)
eset2 <- exprs(celfiles.rma)
par(mfrow=c(1,2))
boxplot(celfiles, col = cols, las = 2, main = "Pre-Normalization")
boxplot(celfiles.rma, col = cols, las = 2, main = "Post-Normalization");
dev.off()
write.table(eset2, file="GSE25861_eset.txt", row.names = TRUE, col.names = TRUE, sep ="\t")
library(limma)
samples <- as.factor(samples)
design <- model.matrix(~0+samples)
colnames(design)
colnames(design) <- c("IUGR", "AGA")
fit = lmFit(eset2, design)
contrast.matrix = makeContrasts("IUGR-AGA", levels = design)
fit2 <- contrasts.fit(fit, contrast.matrix)
fit2 <- eBayes(fit2)
output1 <- topTable(fit2, adjust="BH", coef="IUGR-AGA", number=Inf)
write.table(output1, file="IUGR-AGA.txt", sep="\t", quote=FALSE)
png("IUGR_AGA_volcano.png")
result1 <- read.table("IUGR-AGA.txt", header=TRUE, sep="\t", as.is=TRUE)
with(result1, plot(logFC, -log10(P.Value), pch=20, main="IUGR vs AGA"))
with(subset(result1, P.Value <= 0.05 ), points(logFC, -log10(P.Value), pch=20, col="red"))
with(subset(result1, abs(logFC)>=1), points(logFC, -log10(P.Value), pch=20, col="orange"))
with(subset(result1, P.Value<0.05 & abs(logFC)>=1), points(logFC, -log10(P.Value), pch=20, col="green"))
dev.off()
results$FC <- ifelse(results$logFC > 0,2^results$logFC, (2^abs(results$logFC))* (-1))
a1 <- subset(results, P.Value<.05 & logFC>=1)
a2 <- subset(results, P.Value<.05 & logFC<=-1)
a <- rbind(a1,a2)
a <- a[order(-a$FC),]
a <- na.omit(a)
write.table(a, file="IUGR_test.txt", sep="\t", quote=FALSE)
library(gProfileR)
D1 <- read.table("IUGR_test.txt", header=TRUE, sep="\t", as.is=TRUE)
dim(D1)
D2 <- read.table("annot.txt", header=TRUE, sep="\t", as.is=TRUE)
dim(D2)
D3 <- merge(D1, D2, by="ID")
D3 <- na.omit(D3)
write.table(D3, file="FCGannot_IUGR.txt", row.names = FALSE, col.names = TRUE, sep ="\t")
df <- data.frame(D3$Gene.symbol, D3$FC)
df.avg <- aggregate(. ~ D3.Gene.symbol, data = df, mean)
dim(df.avg)
dim(df)
summary(df)
summary(df.avg)
DEG_ann1 <- df.avg[order(-df.avg$D3.FC),]
write.table(DEG_ann1,"GSE25861_avg_filename.txt",sep="\t",row.names=FALSE)
